src/collective/solr/search.py

```diff
@@ -110,6 +111,12 @@ def search(
             field = schema.get(index, None)
             if field is None or not field.stored:
                 logger.warning('sorting on non-stored attribute "%s"', index)
+        if "rows" in parameters and parameters["rows"] > 1000000000:
+            logger.warning(
+                'The "rows" parameter was set to "%s" and automatically limited to 1000000000 to avoid a Solr java.lang.NumberFormatException',
+                parameters["rows"],
+            )
+            parameters["rows"] = 1000000000
```

Comment:
Since [PEP 515](https://peps.python.org/pep-0515/) included in Python 3.6 one can use `1_000_000_000` formatting